JS debugging by using DevTools (#4392)
magento/devdocs: guides/v2.2/ui_comp_guide/troubleshoot/ui_comp_troubleshoot_js.md

console.log(fieldName.name); // product_form.product_form.product-details.container_name.name
```

## Debugging using Chrome DevTools

All modern browsers support “debugging” – a special UI in developer tools that makes finding and fixing errors much easier. 

### Sources panel UI

> [DevTools] provides a lot of different tools for different tasks, and the **Sources** panel is where we debug JavaScript.

Reviewer: ```suggestion
[DevTools] provides a lot of different tools for different tasks, but the **Sources** panel is where you debug JavaScript.
```

1. Open the required page in Chrome.
2. Turn on developer tools with F12 (Windows, Linux) or Cmd+Opt+I (Mac).

> 3. Select the `Sources` pane.

Reviewer: ```suggestion
3. Click the `Sources` tab.
```

![Sources Pane]({{ site.baseurl }}/common/images/debugging-sources-pane.png)

In the previous image, we can see three zones:

1. The `Resources` zone lists all the files as HTML, JavaScript, CSS.
2. The `Source` zone shows the source code of any selected file.
3. The `Information and control` zone is for debugging.

### Breakpoints

A **breakpoint** is a line of code where the debugger will automatically pause the JavaScript execution process.

> For setting a breakpoint, we should click right on the code line number, like in the next image.

Reviewer: ```suggestion
To set a breakpoint, right-click on the code line number (as shown in the next image).
```

![Breakpoints]({{ site.baseurl }}/common/images/debugging-breakpoints.png)

> We can always find a list of breakpoints in the right pane. That’s useful when we have many breakpoints in various files.

Reviewer: ```suggestion
You can always find a list of breakpoints in the right panel, which is useful when you have many breakpoints in various files.
```
